src/routes/load/container/Load.tsx

```diff
@@ -47,7 +47,7 @@ const Load = (): React.ReactElement => {
   const userAddress = useSelector(userAccountSelector)
 
   const addSafeHandler = (safe: SafeRecordProps) => {
-    dispatch(addSafe(safe))
+    dispatch(addOrUpdateSafe(safe))
```

Comment:
What is the reason to create a new separate action?

Reply:
I could do the check there to see if the safe already exists and make an `update safe` or an `add safe` but I will need to repeat the same logic for creating a safe, so I prefer to add an action that does that, what do you think?

Reply:
This method introduces again some errors on owner handling. I have come to a solution with @fernandomg that should help to update the state correctly.

It's already on development, could it be used here?
https://github.com/gnosis/safe-react/blob/a764a23e660252f5ed71d6c3e22fefde97497d65/src/logic/safe/store/reducer/safe.ts#L47